### PR TITLE
created a docker file to aid dev setup

### PR DESCRIPTION
**Pull request by peterjgrainger**
Not sure if this will help but while I was working on the tests for #97 I was having issues setting up so the easiest way I found was to create a docker container so all the runs were consistent.

Hopefully someone else will find it useful

I'll keep working on the tests :)

### CODE REVIEW

**Comment by kirichkov** [comment]
This indeed can be useful with Linux testing. However, it won't aid with OS X and Windows testing. I'll let @rytilahti decide whether we should pull it. I don't see any harm in having it in the repo.

**Comment by peterjgrainger** [comment]
Docker for Mac/ windows is pretty good.  I'm using a Mac now and it works
great!  I updated the reader with instructions on installation on all
platforms

**Comment by peterjgrainger** [comment]
I mean readme!

**Comment by kirichkov** [comment]
Yes, but underneath Docker for Mac is running a minimal Linux distro. So technically speaking you are still testing under Linux, although it's a Linux box running under a VM on a Mac ;-)

**Comment by peterjgrainger** [comment]
Yeah, hypervisor.  Surprising it works pretty much the same on all platforms, I’m always amazed :)


On 13 Oct 2017, at 09:02, Georgi Kirichkov <notifications@github.com> wrote:

Yes, but underneath Docker for Mac is running a minimal Linux distro. So technically speaking you are still testing under Linux, although it's a Linux box running under a VM on a Mac ;-)

File: README.md [hunk 1]
```diff
@@ -210,4 +210,20 @@ Hue is given in degrees (0-360) and saturation and value in percentage.
...
+Install Docker for your system [Docker for Mac](https://docs.docker.com/docker-for-mac/), [Docker for Windows](https://docs.docker.com/docker-for-windows/), [Docker for Linux](https://docs.docker.com/engine/installation/#server)
+
+#### Build and run
+
+This command will set up the environment and run the tests on demand.
+
+replace `/absolute/path/to/root/of/directory` after the -v flag below with the absolute path to the directory this project is stored in e.g. `/home/bob.smith/pyHS100`
+
+```shell
+docker build . -t pyhs100 && docker run -v /absolute/path/to/root/of/directory/pyHS100/tests:/opt/pyHS100/pyHS100/tests  pyhs100 pytest
```
Comment (rytilahti) [inline README.md]: Considering that the build is being run inside the main directory, `/absolute/path/to/root/of/directory` part could be replaced with `$(PWD)/pyHS100/tests` to make it easier to use.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,50 @@
+####################################################
+# Use python 2.x and the latest version of alpine  #
```
Comment (rytilahti) [inline Dockerfile]: We do not support python2 anymore :-)

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+###################################################
+# Install everything to the path                  #
+###################################################
+RUN python setup.py install
+
+###################################################
+# Run tests                                         #
+###################################################
+CMD pytest
```
Comment (rytilahti) [inline Dockerfile]: Missing a newline at the end, and those "boxes" are also not properly intended.

File: README.md [hunk 4]
```diff
@@ -210,4 +210,20 @@ Hue is given in degrees (0-360) and saturation and value in percentage.
 print(bulb.hsv)
 if bulb.is_color:
    bulb.hsv = (180, 100, 100) # set to cyan
+```
+
+## Development Setup
+
+### Docker
+
+Install Docker for your system [Docker for Mac](https://docs.docker.com/docker-for-mac/), [Docker for Windows](https://docs.docker.com/docker-for-windows/), [Docker for Linux](https://docs.docker.com/engine/installation/#server)
```
Comment (rytilahti) [inline README.md]: Is this part (especially the links) necessary? I would just move the next subchapter here, and mention in the beginning that the system must have a working docker installation.

**Review by rytilahti — CHANGES_REQUESTED** [review]
Looks good, I added a couple of improvement suggestions. I see also no harm for having this in the repository for those who want to use it, even when I personally prefer to use tox&virtualenvs.

File: README.md (hunk 1 above)
Comment (peterjgrainger) [reply README.md]: yeah, I had that at first but then realised it won't work on Windows.

I could give one setup for Windows and one for Mac/Linux.

or I could just ignore Windows users :)

Let me know and I'll do that

Comment (rytilahti) [reply README.md]: I would say that if someone is really developing something on Windows they pretty much know what they are doing, considering how difficult it all feels (to me at least). So it should be fine.

Comment (kirichkov) [reply README.md]: I totally agree with @rytilahti 👍 I'm also in the same boat, I think python (and scripting languages in general) can be developed on Windows, but there's so many tools missing it requires a significant upfront investment to bring your environment en par with a UNIX one.

So yeah, if you're developing Python code on Windows, there's probably a good reason for it, and you should already have a good idea how to find your way around.

**Comment by peterjgrainger** [comment]
believe all comments are fixed now.  Let me know if you need anything else changing

**Review by rytilahti — APPROVED** [review]

**Comment by rytilahti** [comment]
Looks good, thanks!

**Pull request merged**